### ISSUE TITLE
Unable to open Version 1 repositories

### ISSUE THREAD

**Issue by casey** [rust-lang/git2-rs#546]
I get this error when trying to open version 1 git repositories:

```
unsupported repository version 1. Only versions up to 0 are supported.; class=Repository (6)
```

I tried using git2 from github:

```
[dependencies.git2]
git = "https://github.com/rust-lang/git2-rs.git"
```
I thought that perhaps this would work, since I saw that libgit2 added support for V1 repos on Feb 6th:

https://github.com/libgit2/libgit2/commit/06f0230073e0159933aa19d97fce0cbe3b9a3231

And that the version was bumped to 1.0 on March 28th:

https://github.com/libgit2/libgit2/commit/274b2a017db3c3e56959bf78f4dcd8e1d4898257

So I thought that git2 from master would support V1 repos. Cargo tree shows the following:

```
├── git2 v0.13.2 (git+https://github.com/rust-lang/git2-rs.git#c42e8e964b99e193ec3e64619c2be34f262ea0c2)
│   ├── libgit2-sys v0.12.3+1.0.0 (git+https://github.com/rust-lang/git2-rs.git#c42e8e964b99e193ec3e64
```

So it looks like it's using the correct version from master.

Is there anything additional needed to enable support for V1 repos?

And for reference, the repo I'm trying to use is [this one](https://fuchsia.googlesource.com/fuchsia/).

**Comment by alexcrichton** [comment]
It may be the case that commit wasn't included in the 1.0 release? I'm honestly not entirely sure. This may be a better question for libgit2 itself?

**Comment by ehuss** [comment]
I tried `cargo run --example clone https://fuchsia.googlesource.com/fuchsia fucshia` and I got a zlib error:

`error: error reading from the zlib stream; class=Zlib (5)9425)  /  chk   0% (   0/   0)`

Cloning that repo with the git cli, it indicates it is version 0, so I'm not sure why you got a version error. Is the version error maybe misleading to some other problem?

**Comment by casey** [comment]
> It may be the case that commit wasn't included in the 1.0 release? I'm honestly not entirely sure. This may be a better question for libgit2 itself?

I just tried out v1 of libgit2 (commit `7d3c7057f0e774aecd6fc4ef8333e69e5c4873e0`) from a simple C program, and I got the same error. I think I was looking a commit that bumbed the libgit2 version number to 1 in a constant, and not the actual release commit, and that you're right and this didn't actually make it into libgit2 v1.0.

Sorry for the spurious issue!
